packages/react/src/analytics/integrations/GA4/GA4.ts

```diff
@@ -377,11 +380,39 @@ class GA4 extends integrations.Integration<GA4IntegrationOptions> {
       this.processInteractionEvents(commandList, data?.event);
 
       each(commandList, command => {
+        this.attachUniqueEventIdToCommand(command, data);
+
         window.gtag.apply(null, command);
       });
     }
   }
 
+  /**
+   * Attaches the unique event ID generated by the analytics package.
+   *
+   * @param command - The command being processed.
+   * @param data - Event data provided by analytics.
+   *
+   * @returns The modified command.
+   */
+  attachUniqueEventIdToCommand(
+    command: GA4Command,
+    data:
+      | TrackEventData
+      | PageviewEventData
+      | LoadIntegrationEventData
+      | SetUserEventData,
+  ) {
+    const eventProperties = command[2] as Record<string, string>;
```

Comment:
This could be:

```
   const eventProperties = command[2];

    if (eventProperties && typeof eventProperties === 'object') {
      const uniqueEventId = get(
        data,
        'context.event.__uniqueEventId',
      ) as string;

      (eventProperties as Record<string, string>)[OPTION_UNIQUE_EVENT_ID_KEY] =
        uniqueEventId;
    }

    return command;
```

As there is a check already to see if  `eventProperties` is an object, we can take advantage of that and with a cast that is valid inside the if, we can make this type safe.